sql/api/src/main/scala/org/apache/spark/sql/SqlApiConf.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import org.apache.spark.util.SparkClassUtils
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.util.Try
+
+/**
+ * Configuration for all objects that are placed in the `sql/api` project. The normal way of
+ * accessing this class is through `SqlApiConf.get`. If this code is being used with sql/core
+ * then it values are bound to the currently set SQLConf. In connect mode it will default to
+ * hardcoded values.
+ */
+private[sql] trait SqlApiConf {
```

Comment:
no biggie and just dropping a comment. I wonder if we should name it like just `SQLConfiguration` or just `SQLConf` (since `org.apache.spark.sql.internal.SQLConf` at `sql/core` isn't API anyway)

Reply:
Probably not  name as `SQLConf`, as that will triggers a huge amount of refactorings.

Reply:
Is `org.apache.spark.SparkConf` a public API? How about naming it `SparkSQLConf`?

Reply:
`SparkSQLConf` SGTM

Reply:
> `SparkSQLConf` SGTM

+1

Reply:
It really should be an internal thing. It is marked `private[sql]` for this reason. Users should only interact with `RuntimeConf`, and perhaps `SparkSession.Builder` to get/set options.

Reply:
Yeah, if it's internal, I am good with whatever name :-)